pkg/otlp/internal/serializerexporter/exporter.go

```diff
@@ -176,8 +177,30 @@ func (e *exporter) ConsumeMetrics(ctx context.Context, ld pmetric.Metrics) error
 	}
 
 	consumer.addTelemetryMetric(e.hostname)
+	if hasRuntimeMetrics(ld) {
+		consumer.addRuntimeTelemetryMetric(e.hostname)
+	}
 	if err := consumer.Send(e.s); err != nil {
 		return fmt.Errorf("failed to flush metrics: %w", err)
 	}
 	return nil
 }
+
+func hasRuntimeMetrics(ld pmetric.Metrics) bool {
```

Comment:
It would be better for performance to do this check in the call to `e.tr.MapMetrics()` since we're already looping through all of the metrics in that function, but I'm not sure if it's preferable to modify the signature of `MapMetrics` or if there's another pattern we can use with the Translator object.